### PR TITLE
KNL-1575 cookie cluster fix

### PR DESCRIPTION
**Pull request by jayee**
Added throwing away non sakai JSESSIONID cookies

### CODE REVIEW

**Comment by ern** [comment]
This could be reorganized to perform the same checks more than once, i.e. (suffix != null).

**Comment by ottenhoff** [comment]
@ern can you edit this PR in browser per your comment?

File: kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java [hunk 1]
```diff
@@ -1351,15 +1365,25 @@ protected Cookie findCookie(HttpServletRequest req, String name, String suffix)
 		{
 			for (int i = 0; i < cookies.length; i++)
 			{
-				if (cookies[i].getName().equals(name))
+				Cookie cookie = cookies[i];
```
Comment (ern) [inline kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: @jayee does this solve your issue:
```
			for (Cookie cookie : cookies)
			{
				if (StringUtils.equals(cookie.getName(), name))
				{
					if (StringUtils.endsWith(cookie.getValue(), StringUtils.join(DOT, suffix)))
					{
						return cookie;
					}
				}
			}
```

Comment (jayee) [reply kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: Yes, it should work.
The important part about not returning a cookie when no suffix match in value is in place.

Comment (ern) [reply kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: Do you mind testing this first, to see if it addreses it.

Comment (jayee) [reply kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: I'm no longer working with our sakai plattform, but I will ping those who do.

**Pull request closed without merging**